client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useMemo, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { get, extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return get(dataSource, "apply_auto_limit", false);
+}
+
+export default function useAutoLimitFlags(dataSource, query, setQuery) {
+  const isAvailable = isAutoLimitAvailable(dataSource);
+  const [isChecked, setIsChecked] = useState(localOptions.get("ApplyAutoLimit", true));
```

Comment:
```suggestion
  const [isChecked, setIsChecked] = useState(localOptions.get("applyAutoLimit", true));
```

Small thing to be consistent with other options.